crates/rethnet/src/lib.rs

```diff
@@ -37,31 +38,52 @@ struct NodeArgs {
     verbose: u8,
 }
 
-impl TryFrom<NodeArgs> for Config {
-    type Error = anyhow::Error;
-
-    fn try_from(node_args: NodeArgs) -> Result<Config, Self::Error> {
-        Ok(Config {
-            address: SocketAddr::new(node_args.host, node_args.port),
-            rpc_hardhat_network_config: RpcHardhatNetworkConfig {
-                forking: if let Some(json_rpc_url) = node_args.fork_url {
-                    Some(RpcForkConfig {
-                        json_rpc_url,
-                        block_number: node_args.fork_block_number,
-                        http_headers: None,
-                    })
-                } else if node_args.fork_block_number.is_some() {
-                    Err(anyhow!(
-                        "A fork block number can only be used if you also supply a fork URL"
-                    ))?
-                } else {
-                    None
-                },
-            },
-        })
+pub struct ConfigFile {
```

Comment:
Move to a separate file `config.rs` and add documentation

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4149/commits/41350ea5ccf240c7f0ca30e92faa7050826ef7fc